### PR TITLE
Levenshtein distance

### PR DESCRIPTION
**Pull request by ali-abdullah**
Added Levenshtein distance to the program and made a few edits to variable namely:
1. word_2
2. sorting distance 
3. changed the name of levenshtein_distance to edit_distance.

### CODE REVIEW

File: fzysearch/__init__.py [hunk 1]
```diff
@@ -1,3 +1,37 @@
+# Input the two numbers of which the edit distance is to be calculated 
+def levenshtein_distance(word1,word2):
```
Comment (irfansharif) [inline fzysearch/__init__.py]: use `first_word` or `first` and `second_word` or `second` instead, better names.

File: fzysearch/__init__.py [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+        current_row = [previous_row[0] + 1] 
+    
+    # The newest index in the current row will be equal to the smallest value + 1 between
+    # the previous index in the same row, the same row in the previous column
+    # and the previous column in the previous row  
+    # the edit distance will be the value in the last column of the last row 
+        for column in xrange(1,columns):
+            top_box = previous_row[column]
+            left_box = current_row[column - 1]
+            diagnol = previous_row[column - 1]
```
Comment (irfansharif) [inline fzysearch/__init__.py]: `diagonal`

File: fzysearch/__init__.py [hunk 3]
```diff
@@ -15,8 +49,8 @@ def fuzzymatch(input, gallery):
                 # These here are to be used in the two step sorting process,
                 # arranging our results in order of relevance
                 distance_list = []
-                levenshtein_distance = 0
-                sorting_distance = levenshtein_distance
+                edit_distance = 0
```
Comment (irfansharif) [inline fzysearch/__init__.py]: keep this as `levenshtein_distance`

File: fzysearch/__init__.py [hunk 4]
```diff
@@ -34,11 +68,22 @@ def fuzzymatch(input, gallery):
...
+                                # Word_1 the users input 
+                                # Word_2 the words in the list 
+                                # both are passed as parameters into the levenshtein distance function
+                                # the function returns the edit distance
+                                # the shortest number of edits required to change Word_2 into Word_1 
+                                # note the function only considers the characters between the first appearence of the first character in the input string 
+                                # and the last character in the input string
+                                # the edit distance will be used to sort the objects in the list based on relevance
+                                # smaller the edit distance number greater the relevance
                                 word_1 = search
```
Comment (irfansharif) [inline fzysearch/__init__.py]: same thing here, change variable names.

File: fzysearch/__init__.py [hunk 5]
```diff
@@ -1,8 +1,8 @@
 # Input the two numbers of which the edit distance is to be calculated
```
Comment (ridwanmsharif) [inline fzysearch/__init__.py]: Input two words/strings*

File: fzysearch/__init__.py [hunk 6]
```diff
@@ -54,22 +99,34 @@ def fuzzymatch(input, gallery):
...
+                                # and the last character in the input string
+                                # the edit distance will be used to sort the objects in the list based on relevance
+                                # smaller the edit distance number greater the relevance
+                                first_word = search
+                                second_word = library[distance_list[0] : distance_list[-1] + 1]
+                                edit_distance += levenshtein_distance(first_word,second_word)
+                                
+                                sorting_distance += edit_distance
+                                
+                                #print(sorting_distance)
```
Comment (ridwanmsharif) [inline fzysearch/__init__.py]: Not needed anymore

File: fzysearch/__init__.py [hunk 7]
```diff
@@ -34,11 +68,22 @@ def fuzzymatch(input, gallery):
...
```
Comment (ridwanmsharif) [inline fzysearch/__init__.py]: Not needed anymore, needed it just for debugging

**Pull request merged**